test/unit/networking.test.js

```diff
@@ -5,16 +5,96 @@ import sinon from 'sinon';
 import nock from 'nock';
 import assert from 'assert';
 
+import Config from '../../src/core/components/config';
 import Networking from '../../src/networking';
 import { del, get, post } from '../../src/networking/modules/web-node';
 import { keepAlive, proxy } from '../../src/networking/modules/node';
 
+describe('getStandardOrigin', () => {
+  before(() => nock.disableNetConnect());
+  after(() => nock.enableNetConnect());
+
+  describe('if useDomainSharding is unset', () => {
+    it('shouldn\'t generate sharded domains based on custom origin', () => {
+      const config = new Config({ setup: { ssl: true, origin: 'example.pubnubapi.com' } })
+      const networking = new Networking({})
+      networking.init(config);
+
+      const origin = networking.getStandardOrigin();
+
+      assert.equal(origin, "https://example.pubnubapi.com");    
+    });
+
+    it('should generate sharded domains based on default origin', () => {
+      const config = new Config({ setup: { ssl: true } })
+      const networking = new Networking({})
+      networking.init(config);
+
+      const origin = networking.getStandardOrigin();
+
+      const regex = /^https:\/\/ps([0-9](0)?)\.pndsn\.com$/;
+
+      assert(regex.test(origin), `"${origin}" doesn't match the regex ${regex}`);
+    });
+  });
+
+  describe('if useDomainSharding is set to false', () => {
+    it('shouldn\'t generate sharded domains based on custom origin', () => {
+      const config = new Config({ setup: { ssl: true, origin: 'example.pubnubapi.com', useDomainSharding: false } })
+      const networking = new Networking({})
+      networking.init(config);
+
+      const origin = networking.getStandardOrigin();
+
+      assert.equal(origin, 'https://example.pubnubapi.com');
+    });
+
+    it('shouldn\'t generate sharded domains based on default origin', () => {
+      const config = new Config({ setup: { ssl: true, useDomainSharding: false } })
+      const networking = new Networking({})
+      networking.init(config);
+
+      const origin = networking.getStandardOrigin();
+
+      assert.equal(origin, 'https://ps.pndsn.com');
```

Comment:
This seems odd. Do we really want to allow customers to disable domain sharding for ps.pndsn.com?

Reply:
Yes, for example in node.js environments!